### PR TITLE
add sql scripts files in src for sqlthread

### PR DESCRIPTION
**Pull request by cis-muzahid**
add sql scripts files in src for sqlthread,
split sqlthread PR

### CODE REVIEW

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/sql/init_version_3.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+--
+-- Add a new column to the inventory table to store tags.
+--
+
+ALTER TABLE inventory ADD tag blob DEFAULT '';
```
[the remark this answers is not in the document]
Comment (cis-muzahid) [reply src/sql/init_version_3.sql]: add new line at the end of filoe

File: src/sql/init_version_5.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ -- Add a new table: objectprocessorqueue with which to hold objects
+ -- that have yet to be processed if the user shuts down Bitmessage.
+ --
+
+DROP TABLE knownnodes;
+
+CREATE TABLE `objectprocessorqueue` (
+    `objecttype` text,
+    `data` blob,
+UNIQUE(objecttype, data) ON CONFLICT REPLACE
```
[the remark this answers is not in the document]
Comment (cis-muzahid) [reply src/sql/init_version_5.sql]: changes done

**Review by PeterSurda — APPROVED** [review]

**Comment by PeterSurda** [comment]
These files need to be added to deployment lists:
- dpkg
- maybe appimage (not sure if updating dpkg is enough or need additional steps)
- pyinstaller package file
- macos (py2app) package file
- apk build file (don't remember if this is merged yet or not)

**Comment by cis-muzahid** [comment]
> These files need to be added to deployment lists:
> 
>     * `setup.py` (setuptools)
> 
>     * `bitmessagemain.spec` (pyinstaller package file)
> 
>     * `build_osx.py` (macos (py2app) package file)
> 
>     * apk build file, which is probably something like `buildozer.spec`, but it isn't merged yet, it just needs to be updated on other dev branches
> 
> 
> Just repeat the process done with `default.ini` a copule of weeks ago.

Done changes with setup.py, bitmessagemain.specs and src/build_osx.py but how can I add SQL file to `buildozer.spec` I need to help on that

File: packages/pyinstaller/bitmessagemain.spec [hunk 3]
```diff
@@ -72,6 +72,13 @@ a.datas += [
     for file_ in os.listdir(dir_append) if file_.endswith('.ui')
 ]
 
+sql_dir_append = os.path.join(srcPath, 'sql')
```
Comment (PeterSurda) [inline packages/pyinstaller/bitmessagemain.spec:75]: I think you can use  `glob`, that simplifies it.

File: src/build_osx.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 mainscript = ["bitmessagemain.py"]
 
 DATA_FILES = [
-    ('', ['sslkeys', 'images', 'default.ini']),
+    ('', ['sslkeys', 'images', 'default.ini', 'sql/*.sql']),
```
Comment (PeterSurda) [inline src/build_osx.py:12]: you probably need to do
```
('sql', glob('sql/*.sql')),
```

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: packages/pyinstaller/bitmessagemain.spec (hunk 3 above)
Comment (cis-muzahid) [reply packages/pyinstaller/bitmessagemain.spec:75]: Is it like globe('sql') ?

Comment (cis-muzahid) [reply packages/pyinstaller/bitmessagemain.spec:75]: ```sql_dir_append = glob(os.path.join(srcPath, 'sql/*.sql'))```

Comment (cis-muzahid) [reply packages/pyinstaller/bitmessagemain.spec:75]: sql_dir_append = glob(os.path.join(srcPath, 'sql/*.sql'))

File: src/build_osx.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 mainscript = ["bitmessagemain.py"]
 
 DATA_FILES = [
-    ('', ['sslkeys', 'images', 'default.ini']),
+    ('', ['sslkeys', 'images', 'default.ini', glob('sql/*.sql')]),
```
Comment (PeterSurda) [inline src/build_osx.py:12]:
```
('', ['sslkeys', 'images', 'default.ini']),
('sql', glob('sql/*.sql')]),
(...)
```

File: packages/pyinstaller/bitmessagemain.spec [hunk 6]
```diff
@@ -72,6 +73,13 @@ a.datas += [
     for file_ in os.listdir(dir_append) if file_.endswith('.ui')
 ]
 
+sql_dir_append = glob(os.path.join(srcPath, 'sql/*.sql'))
```
Comment (PeterSurda) [inline packages/pyinstaller/bitmessagemain.spec:76]: this whole section can be replaced with glob, like in the other files

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/build_osx.py (hunk 4 above)
Comment (cis-muzahid) [reply src/build_osx.py]: done

File: packages/pyinstaller/bitmessagemain.spec (hunk 6 above)
Comment (cis-muzahid) [reply packages/pyinstaller/bitmessagemain.spec:76]: Is this like?

```a.datas += os.path.join(srcPath, 'sql'), glob(os.path.join(srcPath, 'sql/*.sql')
```

File: packages/pyinstaller/bitmessagemain.spec [hunk 7]
```diff
@@ -72,6 +73,8 @@ a.datas += [
     for file_ in os.listdir(dir_append) if file_.endswith('.ui')
 ]
 
+a.datas += ('sql', glob('sql/*.sql'), 'DATA')
```
Comment (PeterSurda) [inline packages/pyinstaller/bitmessagemain.spec:76]: how about
```
a.datas += [('sql', glob('sql/*.sql'), 'DATA')]
```

**Review by PeterSurda — CHANGES_REQUESTED** [review]
- remove `import glob`
- squash

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**